dvc/config.py

```diff
@@ -101,6 +101,13 @@ class RelPath(str):
     "shared": All(Lower, Choices("group")),
     Optional("slow_link_warning", default=True): Bool,
 }
+HTTP_COMMON = {
+    "basic_auth": Bool,
+    "digest_auth": Bool,
```

Comment:
@shcheklein Noticed that we might want to generalise this to support commonly-used custom headers. So how about we adjust this to:
```
"auth": All(Lower, Choices("basic", "digest", "custom")),   # with potential to expand it to oauth, kerberos, etc
"custom_header": str,
"user": str,
"password": str,
"ask_password": Bool,
}
```
and then for `custom` set something like `r.headers[self.custom_header] = self.password`. What do you think @pmrowla ?

Reply:
Sounds like a reasonable solution to me

Reply:
One question, should we just separate the ability to provide custom headers from the auth entirely?

i.e. only allow `Choices("basic", "digest")` for `auth`, and have a separate `custom_headers` config option where users can supply a list of `"Header: Value"` strings that will be added to any requests for that remote?

Reply:
I think I'm fine with either since both options solve the problem. In general it's a choice of "being flexible but and bit lower level" vs "less flexible and more explicit".

A few things to consider:

- it means `auth` type will be become optional, right?
- also how would the `dvc remote modify` command look like? can we come with an example to see if it's not too bloated and more or less easy to write?
- is it fine with us that the whole option will have to go into the local config vs an ability to split header name and password into two configs and manage separately (I can't come up with a case where someone would need this to be honest)

what do you think @pmrowla ?

Reply:
My initial line of thinking was that in .dvc/config it would look something like
```
['remote "http-custom"']
    url = http://localhost:5000/
    custom_header = My-Header1: value1
    custom_header = My-Header2: value2
    ...
```
And then in the http remote, we would parse all entries for `custom_header`. 

But thinking about it some more, I don't think the benefits of supporting this would be worth the added complexity of making the CLI tools handle options that can be specified multiple times (especially once you get into how you would unset specific values from the CLI).

I'll just go ahead and proceed with what @efiop originally suggested

Reply:
@pmrowla sounds good! It might make sense to rename the `custom_header` into `custom_auth_header`? to make it more explicit again, and keep us an option of adding that list you mentioned.